### PR TITLE
Feat/redirect to dis moi app

### PR DESCRIPTION
**Pull request by felix-lambert**
When i enable accessibility service the dismoi app should be launched automatically

### CODE REVIEW

**Comment by lutangar** [comment]
Too many styles diff, quite difficult to review in this state, maybe chang the target branche and rebase?

**Comment by felix-lambert** [comment]
Maybe review first: 
https://github.com/dis-moi/mobile_poc/pull/31
Then:
https://github.com/dis-moi/mobile_poc/pull/35
And then:
https://github.com/dis-moi/mobile_poc/pull/36

**Comment by lutangar** [comment]
> Maybe review first:

If we do this, not only I have to review but we'll have to merge before moving on to the next review, I'd rather change targeted branch if they are dependent on each others

**Comment by felix-lambert** [comment]
#36 is an evolution of #35 so didn't want to put on the same target branch to avoid conflicts.
#31 is independent so has the same target branch as #35

File: android/app/src/main/java/com/dismoi/scout/Floating/Layout/Base.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  Coordinator getLayoutCoordinator() {
+    return layoutCoordinator;
+  }
+
+  public void setWindowManager(WindowManager windowManager) {
+    this.windowManager = windowManager;
+  }
+
+  public void create(WindowManager getWindowManager, WindowManager.LayoutParams layoutParams, Coordinator layoutCoordinator) {
```
Comment (lutangar) [inline android/app/src/main/java/com/dismoi/scout/Floating/Layout/Base.java:25]: This looks like a constructor and the job of a constructor, so it's should probably one.

File: android/app/src/main/java/com/dismoi/scout/Floating/Layout/Bubble.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  }
+
+  private void move(float deltaX, float deltaY) {
+    getViewParams().x += deltaX;
+    getViewParams().y += deltaY;
+    windowManager.updateViewLayout(this, getViewParams());
+  }
+
+
+  private class MoveAnimator implements Runnable {
```
Comment (lutangar) [inline android/app/src/main/java/com/dismoi/scout/Floating/Layout/Bubble.java:166]: I'm pretty sure I arleady reviewed that somewhere :thinking:

File: android/app/src/main/java/com/dismoi/scout/Floating/Layout/Trash.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import android.animation.AnimatorInflater;
+import android.animation.AnimatorSet;
+import android.content.Context;
+import android.os.Vibrator;
+import android.util.AttributeSet;
+
+import com.dismoi.scout.R;
+
+public class Trash extends Base {
+  public static final int VIBRATION_DURATION_IN_MS = 70;
```
Comment (lutangar) [inline android/app/src/main/java/com/dismoi/scout/Floating/Layout/Trash.java:12]: Love those meaningful namings <3

File: android/build.gradle [hunk 4]
```diff
@@ -12,13 +12,31 @@ buildscript {
...
+
+        classpath('com.android.tools.build:gradle:4.1.2')
+        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.0'
+        classpath 'com.github.dcendents:android-maven-gradle-plugin:1.4.1'
+
+        
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
+
+        // classpath 'com.github.dcendents:android-maven-plugin:1.2'
```
Comment (lutangar) [inline android/build.gradle:24]: Still relevant?

File: android/build.gradle [hunk 5]
```diff
@@ -12,13 +12,31 @@ buildscript {
...
+        // classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.3'
     }
 }
 
+
 allprojects {
+
+    // allprojects {
```
Comment (lutangar) [inline android/build.gradle:32]: Still relevant?

File: android/settings.gradle [hunk 6]
```diff
@@ -1,5 +1,5 @@
 rootProject.name = 'dismoi'
-include ':react-native-floating-bubble'
-project(':react-native-floating-bubble').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-floating-bubble/android')
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
 include ':app'
+
+//, ':bubbles'
```
Comment (lutangar) [inline android/settings.gradle:5]: Still relevant?

**Review by lutangar — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+    ImageButton imageButton = (ImageButton) messageDisMoiView.findViewById(R.id.close);
+    imageButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        sendEventToReactNative("floating-dismoi-message-press");
+      }
+    });
+
+    messageDisMoiView.setOnBubbleRemoveListener(new Message.OnBubbleRemoveListener() {
```
[the remark this answers is not in the document]
Comment (felix-lambert) [reply android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java:133]: I'm not sure I'm getting this very well. Can you show me some code example?

File: android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+
+public class FloatingModule extends ReactContextBaseJavaModule {
+
+  private Manager bubblesManager;
+  private Manager messagesManager;
+  private final ReactApplicationContext reactContext;
+  private Bubble bubbleDisMoiView;
+  private Message messageDisMoiView;
+
+  public FloatingModule(ReactApplicationContext reactContext) {
```
[the remark this answers is not in the document]
Comment (felix-lambert) [reply android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java:34]: Code example or new PR would be great

File: android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (lutangar) [reply android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java:133]: I don't have an example but I supposed  (maybe wrongly) :
```
      public void onBubbleRemoved(Message bubble) {
        messageDisMoiView = null;
        sendEventToReactNative("floating-dismoi-message-remove");
      }
```
...could possibly be defined once in the class.

If not posible just ignored this.

File: android/app/src/main/java/com/dismoi/scout/Floating/FloatingService.java [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+            break;
+          }
+        }
+      }
+    });
+  }
+
+  private WindowManager getWindowManager() {
+    if (windowManager == null) {
+      windowManager = (WindowManager)getSystemService(WINDOW_SERVICE);
```
[the remark this answers is not in the document]
Comment (lutangar) [reply android/app/src/main/java/com/dismoi/scout/Floating/FloatingService.java:82]: Usually we instantiate once in constructor, so you don't have to check if `windowManager == null` here.
But might be different usecase here, ignore my comment if not relevant.

File: android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (lutangar) [reply android/app/src/main/java/com/dismoi/scout/Floating/FloatingModule.java:34]: I'd love to dig on this, but I probably won't have the time, so just ignore this.

**Review by lutangar — APPROVED** [review]

**Pull request merged**